### DISCUSSION TITLE
f951: Fatal Error: Reading module ‘module_date_utilities’ at line 1 column 2: Unexpected EOF

### DISCUSSION THREAD

**Discussion by shitenglong** [NCAR/hrldas#171] [Q&A] [answered]
When I executed make I got the following error：
(cd MPP; make -f Makefile.NoahMP)
make[1]: Entering directory '/home/shitl/hrldas/hrldas/MPP'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/hrldas/MPP'
(cd Utility_routines;		make)
make[1]: Entering directory '/home/shitl/hrldas/hrldas/Utility_routines'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/hrldas/Utility_routines'
(cd ../noahmp/utility;          make)
make[1]: Entering directory '/home/shitl/hrldas/noahmp/utility'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/noahmp/utility'
(cd ../noahmp/src;              make)
make[1]: Entering directory '/home/shitl/hrldas/noahmp/src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/noahmp/src'
(cd ../noahmp/drivers/hrldas;   make)
make[1]: Entering directory '/home/shitl/hrldas/noahmp/drivers/hrldas'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/noahmp/drivers/hrldas'
(cd ../urban/wrf;               make)
make[1]: Entering directory '/home/shitl/hrldas/urban/wrf'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/shitl/hrldas/urban/wrf'
(cd IO_code;			make)
make[1]: Entering directory '/home/shitl/hrldas/hrldas/IO_code'

rm -f module_hrldas_netcdf_io.f90
cpp -P -traditional -DMPP_LAND -D_GFORTRAN_  -D_HRLDAS_OFFLINE_  -I/home/shitl/Build_WRF/LIBRARIES/netcdf/include module_hrldas_netcdf_io.F > module_hrldas_netcdf_io.f90
gfortran -o module_hrldas_netcdf_io.o -c -g -fconvert=big-endian -fbounds-check -fno-range-check  -ffree-form  -ffree-line-length-none -I../MPP -I../Utility_routines -I/home/shitl/Build_WRF/LIBRARIES/netcdf/include  module_hrldas_netcdf_io.f90
f951: Fatal Error: Reading module ‘module_date_utilities’ at line 1 column 2: Unexpected EOF
compilation terminated.
make[1]: *** [Makefile:40: module_hrldas_netcdf_io.o] Error 1
make[1]: Leaving directory '/home/shitl/hrldas/hrldas/IO_code'
make: *** [Makefile:12: all] Error 2
##=========================================
The compilation environment is as follows：
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.3)
Can someone help me?

**Comment by cenlinhe** [comment, marked as answer]
This is probably because of your computer environment setup. Maybe you could try to use our Noah-MP Docker container which includes all the compiler and packages you need to run the model. you can follow this instruction to download and set up docker container: https://github.com/NCAR/hrldas/blob/develop/tutorial/NoahMP_docker_container_setup.pdf
Here is a follow-up tutorial after you set up the docker container: https://github.com/NCAR/hrldas/blob/master/tutorial/Note6_AMS2024_NoahMP_short_course.ipynb

**Reply by 12wlj** [reply]
请问解决问题了吗，我也遇到这个问题了

**Reply by shitenglong** [reply]
用Docker吧，自己装的环境兼容不了

**Reply by cenlinhe** [reply]
Hi, this is a problem with your computer linux compiler and software environment setup, not related to model code. So we are not able to do anything from our end. You can use our docker container mentioned above which provides all the required environment setup to run the model. HRLDAS/NoahMP can be run in typical linux/unix machine/environment with fortran intel/gnu/pgi compiler and netcdf software.

**Reply by 12wlj** [reply]
我试一试，谢谢了。
